src/mon/OSDMonitor.cc

```diff
@@ -970,31 +970,192 @@ void OSDMonitor::encode_pending(MonitorDBStore::TransactionRef t)
     tmp.apply_incremental(pending_inc);
 
     if (tmp.require_osd_release >= CEPH_RELEASE_LUMINOUS) {
-      // set or clear full/nearfull?
+      // remove any legacy osdmap nearfull/full flags
+      {
+        dout(10) << __func__ << " clearing legacy osdmap nearfull/full flag"
+                 << dendl;
+        remove_flag(CEPH_OSDMAP_NEARFULL);
+        remove_flag(CEPH_OSDMAP_FULL);
+      }
+
       int full, backfill, nearfull;
       tmp.count_full_nearfull_osds(&full, &backfill, &nearfull);
```

Comment:
since we will call teh get_full_pool_ids etc for any of these that are nonzero, it probably makes sense to make those optional (or requried?) args for thsi method and populate it all at the same time.  otherwise we are iterating over the list(s) so many times.  I'm worried these are pretty inefficient search methods!